src/main/java/org/wso2/securevault/secret/repository/VaultSecretRepositoryProvider.java

```diff
@@ -0,0 +1,186 @@
+                        Object repositoryImpl = repositoryClass.newInstance();
+
+                        if (repositoryImpl instanceof SecretRepository) {
+                            Properties repositoryProperties = filterConfigurations(configurationProperties, repo);
+                            ((SecretRepository) repositoryImpl).init(repositoryProperties, providerType);
+                            vaultRepositoryMap.put(repo, (SecretRepository) repositoryImpl);
+                        }
+                    } catch (ClassNotFoundException e) {
+                        throw new SecureVaultException(
+                                "A SecretRepository cannot be found for class name: " + repositoryClassName, e);
+                    } catch (IllegalAccessException e) {
+                        throw new SecureVaultException(
+                                "Error creating an instance, Method does not have access to the class: " +
+                                        repositoryClassName, e);
+                    } catch (InstantiationException e) {
+                        throw new SecureVaultException(
+                                "Error creating an instance from class: " + repositoryClassName, e);
+                    }
+                }
+            }
+        }
+        return vaultRepositoryMap;
+    }
+
+    /**
+     * Return the properties for a provided repository.
+     *
+     * @param configProperties All the properties under secret configuration file.
+     * @param repository       Repository listed under the vault provider.
+     * @return Filtered properties.
+     */
+    private static Properties filterConfigurations(Properties configProperties, String repository) {
+
+        Properties filteredProps = new Properties();
+        StringBuilder propertyKeyPrefix = new StringBuilder()
+                .append(repository)
+                .append(DOT)
+                .append(PROPERTIES);
+
+        configProperties.forEach((propKey, propValue) -> {
+            if (propKey.toString().contains(propertyKeyPrefix)) {
+                filteredProps.put(propKey, propValue);
+            }
+        });
+        return filteredProps;
+    }
+
+    /**
+     * Util method for getting property values from the secret-conf file.
+     *
+     * @param secretConfigProps All the properties under secret configuration file.
+     * @param propName          Name of the property.
+     * @return Returns the value for the give property.
+     */
+    private static String getPropertiesFromSecretConfigurations(Properties secretConfigProps, String propName) {
+
+        return MiscellaneousUtil.getProperty(secretConfigProps, propName, null);
+    }
+
+    /**
+     * Validate the property value to avoid the processing of null values.
+     *
+     * @param propValue Value of the required property.
+     * @return Return true if not null.
+     */
+    private static boolean isPropValueValidated(String propValue) {
```

Comment:
shouldn't this be isPropertyValueValied ?

Reply:
Changed